src/components/targets/createTarget/selectDropdown.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react";
+import 'components/targets/createTarget/selectDropdown.css'
+
+import football from 'assets/images/football.png'
+import music from 'assets/images/music.png'
+import food from 'assets/images/food.png'
+import travel from 'assets/images/travel.png'
+
+interface SelectDropDownProps {
+  options: { label: string; value: string }[];
+}
+
+interface ImgHash {
+  [key: string]: string;
+}
+
+const imgHash: ImgHash = {
+  'football': football,
+  'music': music,
+  'food': food,
+  'travel': travel
+}
+
+const SelectDropDown = ({ options, ...props }: SelectDropDownProps) => {
+  const handleSelectChange = (event: any) => {
+    const option = event.target.textContent;
+    let selectPlaceholder = document.querySelectorAll('.selectdropdown--placeholder')[0];
+    selectPlaceholder.textContent = option;
+  };
+
+  const changeMenuVisibility = (event: any) => {
+    let visibility  = event.target.parentElement.nextSibling.style.visibility;
+    const menu = event.target.parentElement.nextSibling;
+
+    if(visibility === 'visible') {
+      menu.style.visibility = 'hidden'
+    }else{
+      menu.style.visibility = 'visible'
+    }
+  }
+
+  return (
+    <div>
+      <div className="target--form-input selectdropdown" onClick={changeMenuVisibility}
+      >
+        <a href='#' className='selectdropdown--placeholder'>
+          What do you want to talk about?
+        </a>
+      </div>
+        
+      <div className="selectdropdown--options">
+        <div>
+          {        
+            options.map((option) => (                  
+                <div
+                  key={option.value}
+                  className='selectdropdown--topic-option'
+                  onClick={handleSelectChange}
+                  
+                >            
+                  <div className='selectdropdown--option-img-section'>
+                    <img src={imgHash[option.label.toString()]} className='selectdropdown--option-img'/>
+                  </div>
+                  <div className='selectdropdown--option-label'>
+                    {option.label}
+                  </div>
+                </div>
+              )
+            )
+          }
```

Comment:
And then you can do:
```suggestion
          { areOptionsVisibile &&        
            options.map((option) => (                  
                <div
                  key={option.value}
                  className='selectdropdown--topic-option'
                  onClick={handleSelectChange}
                  
                >            
                  <div className='selectdropdown--option-img-section'>
                    <img src={imgHash[option.label.toString()]} className='selectdropdown--option-img'/>
                  </div>
                  <div className='selectdropdown--option-label'>
                    {option.label}
                  </div>
                </div>
              )
            )
          }
```

Reply:
I tried this but it breaks the space too. I only need to change visibility without affecting the space occupied by the element, I think in this occasion it does not work 🤔